### PR TITLE
dvc: do not set cache dir if it's not a dvc repo

### LINKED ISSUE #3328: `dvc` commands fail on non-DVC repository with an exception
```
$ cd "$(mktemp -d)" && dvc
ERROR: the following arguments are required: COMMAND
usage: dvc [-q | -v] [-h] [-V] COMMAND ...

Data Version Control

optional arguments:
  -q, --quiet    Be quiet.
  -v, --verbose  Be verbose.
  -h, --help     Show this help message and exit.
  -V, --version  Show program's version.

Available Commands:
  COMMAND        Use `dvc COMMAND --help` for command-specific help.
    init         Initialize DVC in the current directory.
    get          Download file or directory tracked by DVC or by Git.
    get-url      Download or copy files from URL.
    destroy      Remove DVC-files, local DVC config and data cache.
    add          Track data files or directories with DVC.
    remove       Remove DVC-file outputs.
    move         Rename or move a DVC controlled data file or a directory.
    unprotect    Unprotect data files or directories.
    run          Generate a stage file from a command and execute the command.
    repro        Check for changes and reproduce stages and dependencies.
    pull         Pull data files from a DVC remote storage.
    push         Push data files to a DVC remote storage.
    fetch        Fetch data files from a DVC remote storage.
    status       Show changed stages, compare local cache and a remote storage.
    gc           Garbage collect unused objects from cache or remote storage.
    import       Download file or directory tracked by DVC or by Git into the workspace, and track it.
    import-url   Download or copy file from URL and take it under DVC control.
    config       Get or set config options.
    checkout     Checkout data files from cache.
    remote       Manage remote storage configuration.
    cache        Manage cache settings.
    metrics      Commands to add, manage, collect and display metrics.
    install      Install DVC git hooks into the repository.
    root         Relative path to the repository's directory.
    list         List files and DVC outputs in the repo.
    lock         Lock DVC-files.
    unlock       Unlock DVC-files.
    pipeline     Manage pipelines.
    commit       Save changed data to cache and update DVC-files.
    diff         Compare two versions of the DVC repository.
                 Shows the list of paths added, modified, or deleted
    version      Show DVC version and system/environment information.
    update       Update data artifacts imported from other DVC repositories.

Having any troubles? Hit us up at https://dvc.org/support, we are always happy to help!
Traceback (most recent call last):
  File "/home/saugat/repos/iterative/dvc/.env/py36/bin/dvc", line 12, in <module>
    sys.exit(main())
  File "/home/saugat/repos/iterative/dvc/dvc/main.py", line 89, in main
    if analytics.is_enabled():
  File "/home/saugat/repos/iterative/dvc/dvc/analytics.py", line 57, in is_enabled
    Config(validate=False).get("core", {}).get("analytics", "true")
  File "/home/saugat/repos/iterative/dvc/dvc/config.py", line 223, in __init__
    self.load(validate=validate)
  File "/home/saugat/repos/iterative/dvc/dvc/config.py", line 282, in load
    self["cache"]["dir"] = os.path.join(self.dvc_dir, "cache")
  File "/home/saugat/.pyenv/versions/3.6.6/lib/python3.6/posixpath.py", line 80, in join
    a = os.fspath(a)
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

### PR DESCRIPTION
**Pull request by skshetry**
* [x] ❗ Have you followed the guidelines in the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) list?

* [x] 📖 Check this box if this PR **does not** require [documentation](https://dvc.org/doc) updates, or if it does **and** you have created a separate PR in [dvc.org](https://github.com/iterative/dvc.org) with such updates (or at least opened an issue about it in that repo). Please link below to your PR (or issue) in the [dvc.org](https://github.com/iterative/dvc.org) repo.

* [x] ❌ Have you checked DeepSource, CodeClimate, and other sanity checks below? We consider their findings recommendatory and don't expect everything to be addressed. Please review them carefully and fix those that actually improve code or fix bugs.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

Fixes #3328

### CODE REVIEW

File: dvc/config.py [hunk 1]
```diff
@@ -277,8 +277,8 @@ def load(self, validate=True):
         self.clear()
         self.update(conf)
 
-        # Add resolved default cache.dir
-        if not self["cache"].get("dir"):
+        # Add resolved defa1ult cache.dir
```
Comment (skshetry) [inline dvc/config.py]: oops, typo
```suggestion
        # Add resolved default cache.dir
```

File: dvc/config.py [hunk 2]
```diff
@@ -277,8 +277,8 @@ def load(self, validate=True):
...
+        if not self["cache"].get("dir") and self.dvc_dir:
```
Comment (skshetry) [inline dvc/config.py]: I'm really not sure if we should load the configs even if it's not a DVC directory.

Comment (efiop) [reply dvc/config.py]: @skshetry Could you elaborate please? You mean that we should call it on-premise when some field in config is being accessed?

Comment (skshetry) [reply dvc/config.py]: I mean, should we try to load config if it's not a DVC repository? Currently, this function that we are discussing loads from all the local, system and global configs even though `self.dvc_dir` is not set (i.e. it's not a DVC dir). What I am asking is, are those configs used anywhere when it's not a DVC dir?

Comment (skshetry) [reply dvc/config.py]: Is there any point to try to load those?

Comment (efiop) [reply dvc/config.py]: @skshetry Ah, got it. Yes, those configs (global and system) are indeed used. For example in analytics.py. Imagine running `dvc  get` outside of dvc repo, where dvc will try to send analytics event and so needs to check if analytics are disabled in the config or not.

Comment (skshetry) [reply dvc/config.py]: Got it, thanks.

**Comment by skshetry** [comment]
Let's fail with the test for now. After it fails, I'll fix a push in another PR.

**Comment by skshetry** [comment]
Failed nicely. CI log:
```
=================================== FAILURES ===================================
_______________ test_config_loads_without_error_for_non_dvc_repo _______________
[gw3] linux -- Python 3.7.1 /home/travis/virtualenv/python3.7.1/bin/python
tmp_dir = PosixTmpDir('/tmp/pytest-of-travis/pytest-0/popen-gw3/test_config_loads_without_erro0')
    def test_config_loads_without_error_for_non_dvc_repo(tmp_dir):
        # regression testing for https://github.com/iterative/dvc/issues/3328
>       Config(validate=True)
/home/travis/build/iterative/dvc/tests/func/test_config.py:108: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/travis/build/iterative/dvc/dvc/config.py:223: in __init__
    self.load(validate=validate)
/home/travis/build/iterative/dvc/dvc/config.py:282: in load
    self["cache"]["dir"] = os.path.join(self.dvc_dir, "cache")
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
a = None, p = ('cache',)
    def join(a, *p):
        """Join two or more pathname components, inserting '/' as needed.
        If any component is an absolute path, all previous path components
        will be discarded.  An empty last part will result in a path that
        ends with a separator."""
>       a = os.fspath(a)
E       TypeError: expected str, bytes or os.PathLike object, not NoneType
/home/travis/virtualenv/python3.7.1/lib/python3.7/posixpath.py:80: TypeError
_______________ test_plain_dvc_command_not_fail_on_non_dvc_repo ________________
[gw1] linux -- Python 3.7.1 /home/travis/virtualenv/python3.7.1/bin/python
tmp_dir = PosixTmpDir('/tmp/pytest-of-travis/pytest-0/popen-gw1/test_plain_dvc_command_not_fai0')
git_dir = PosixTmpDir('/tmp/pytest-of-travis/pytest-0/popen-gw1/git-erepo0')
    def test_plain_dvc_command_not_fail_on_non_dvc_repo(tmp_dir, git_dir):
        # regression test for https://github.com/iterative/dvc/issues/3328
>       assert main([]) == 0
E       assert 254 == 0
E         -254
E         +0

/home/travis/build/iterative/dvc/tests/func/test_cli.py:184: AssertionError
```

**Pull request merged**